### PR TITLE
Topic info editor improvement and verifier

### PR DESCRIPTION
**Pull request by Aesylwinn**
I still need to go over the acceptable values for the verifier and the autofill feature for conditions in the editor. Also, I need to double check that my wrapper is reading and writing the data correctly.

Other than what I mentioned above, the result is pretty much what I was thinking of. There are some major changes to how the editor works. Let me know what you think.

### CODE REVIEW

**Comment by Aesylwinn** [comment]
Good to know. I'll go ahead and change it for now. Would that be related to feature 890?

File: apps/opencs/model/world/infoselectwrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,893 @@
...
+        // Integer
+        case Function_Journal:
+        case Function_Item:
+        case Function_Dead:
+        case Function_RankLow:
+        case Function_RankHigh:
+        case Function_RankRequirement:
+        case Function_Reputation:
+        case Function_PcReputation:
+        case Function_PcLevel:
```
Comment (Aesylwinn) [inline apps/opencs/model/world/infoselectwrapper.cpp]: Should I change the valid range for player/actor level to be greater or equal to 1?

File: apps/opencs/model/world/infoselectwrapper.cpp [hunk 2]
```diff
@@ -0,0 +1,893 @@
...
+        case Function_PcStrength:
+        case Function_PcBlock:
+        case Function_PcArmorer:
+        case Function_PcMediumArmor:
+        case Function_PcHeavyArmor:
+        case Function_PcBluntWeapon:
```
Comment (Aesylwinn) [inline apps/opencs/model/world/infoselectwrapper.cpp]: For player skills, should I change the valid range to be 5 or greater?

File: apps/opencs/model/world/infoselectwrapper.cpp [hunk 3]
```diff
@@ -0,0 +1,893 @@
...
+        case Function_RankLow:
+        case Function_RankHigh:
+        case Function_Reputation:
+        case Function_PcReputation:
+        case Function_Journal:
+            return std::pair<int, int>(IntMin, IntMax);
+
+        case Function_Item:
+        case Function_Dead:
+        case Function_PcLevel:
```
Comment (Aesylwinn) [inline apps/opencs/model/world/infoselectwrapper.cpp]: For player/actor level, should I change the valid range to be 1 or greater?

File: apps/opencs/model/world/infoselectwrapper.cpp [hunk 4]
```diff
@@ -0,0 +1,893 @@
...
+        case Function_PcStrength:
+        case Function_PcBlock:
+        case Function_PcArmorer:
+        case Function_PcMediumArmor:
+        case Function_PcHeavyArmor:
+        case Function_PcBluntWeapon:
+        case Function_PcLongBlade:
```
Comment (Aesylwinn) [inline apps/opencs/model/world/infoselectwrapper.cpp]: For skills, should I change the valid range to be 5 or greater?

**Comment by Aesylwinn** [comment]
Other than my line comments, I consider the topic info verifier to be complete.

**Comment by zinnschlag** [comment]
I suggest to ask these questions on the forum, because I am not quite sure which values would work here.

**Comment by Aesylwinn** [comment]
It's a bit of a hack, but I got the auto completion working. Perhaps I should add another class that inherits from CommandDelegate to accomplish this?

**Comment by Aesylwinn** [comment]
I re-added the journal verifier. I changed the name of the class and some of the error messages.

**Comment by zinnschlag** [comment]
Works nicely. I would consider this feature complete. Is there anything you want to add or should we go ahead with the merge?

**Comment by Aesylwinn** [comment]
You can go ahead with the merge.

**Comment by zinnschlag** [comment]
Merged. Seems we finally have most of the verifiers out of the way. Let me know if you need help finding a new task.

**Comment by psi29a** [comment]
yay! \o/

Thank you for this.

**Comment by Aesylwinn** [comment]
:)
@zinnschlag I'm thinking of doing one of the rendering tasks if there's not too many people working on those. If you have a suggestion for a good one to start with, I'd appreciate it. Also, if you think a different area needs some attention, I can work on that instead.

**Comment by zinnschlag** [comment]
The cell marker task is done (minus the popup menu). Everything else is still available. There is of course also the camera task, which is rather important, but also big and a bit tricky.

**Pull request merged**